flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -491,41 +492,9 @@ static ContainerLaunchContext createTaskExecutorContext(
 			flinkJar = registerLocalResource(fs, remoteJarPath);
 		}
 
-		// register conf with local fs
-		final LocalResource flinkConf;
```

Comment:
Q: This seems to be a relatively small content (especially if we consider this can be constructed as a Stringify argument using the `getDynamicProperties` code. Can I assume that the main time saver here is the sync interaction with the DFS during `setupLocalResource`?

If the above assumption is true, would changing the NMClient / AMRMClient to use async resolve the issue already?

Reply:
I second @walterddr here that wondering how flink-conf.yaml becomes a bottleneck of TM deployment.

Reply:
Uploading a taskmanager-conf.yaml to hdfs will take 50ms in many production hdfs cluster. If we receive more that 1000 containers, this will take long time in `YarnResourceManager` main thread. In addition, all the {uuid}-taskmanager-conf.yaml are same and have very small difference with flink config uploaded by client.

So i suggest to remove the io operation in start TaskExecutor. About `NMClientAsync`, it is another optimazion to eliminate the rpc blocking call in `YarnResourceManager` main thread.